File: src-old/cucumber/java/io/tripled/social/client/CucumberOutput.java
package io.tripled.social.client;

import io.tripled.social.client.presentation.Output;

import java.util.concurrent.BlockingDeque;
import java.util.concurrent.LinkedBlockingDeque;

import static org.hamcrest.Matchers.equalTo;
import static org.junit.Assert.assertThat;

class CucumberOutput implements Output {

  private final BlockingDeque<String> outputs;

  CucumberOutput() {
    this.outputs = new LinkedBlockingDeque<>();
  }

  @Override
  public void print(String message) {
    String[] split = message.split("\\n");
    for (String part : split) {
      this.outputs.offer(part);
    }
  }

  void assertContains(String... lines) {
    while (outputs.size() < lines.length) {
      try {
        Thread.sleep(5);
      } catch (InterruptedException e) {
        e.printStackTrace();
      }
    }

    for (String line : lines) {
      String resultLine = outputs.poll();
      assertThat(resultLine, equalTo(line));
    }
  }
}

File: src/test/kotlin/io/tripled/acceptance/CucumberOutput.kt
package io.tripled.acceptance

import io.tripled.social.client.infrastructure.cli.Output
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import java.util.concurrent.BlockingDeque
import java.util.concurrent.LinkedBlockingDeque

internal class CucumberOutput : Output {
    private val outputs: BlockingDeque<String> = LinkedBlockingDeque()

    override fun print(message: String) {
        val split = message.split("\\n".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        for (part in split) {
            this.outputs.offer(part)
        }
    }

    fun assertContains(vararg lines: String?) {
        while (outputs.size < lines.size) {
            try {
                Thread.sleep(5)
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }
        }

        for (line in lines) {
            val resultLine = outputs.poll()
            MatcherAssert.assertThat<String?>(resultLine, Matchers.equalTo<String?>(line))
        }
    }
}
